addons/website_event_exhibitor/controllers/exhibitor.py

```diff
@@ -77,16 +78,22 @@ def _event_exhibitors_get_values(self, event, **searches):
         sponsor_types = sponsors_all.mapped('sponsor_type_id')
         sponsor_countries = sponsors_all.mapped('partner_id.country_id').sorted('name')
         # organize sponsors into categories to help display
-        sponsor_categories = dict()
+        sponsor_categories_dict = defaultdict(request.env['event.sponsor'].sudo)
+        sponsor_categories = []
+        is_event_user = request.env.user.has_group('event.group_event_user')
         for sponsor in sponsors:
-            if not sponsor_categories.get(sponsor.sponsor_type_id):
-                sponsor_categories[sponsor.sponsor_type_id] = request.env['event.sponsor'].sudo()
-            sponsor_categories[sponsor.sponsor_type_id] |= sponsor
-        sponsor_categories = [
-            dict({
+            sponsor_categories_dict[sponsor.sponsor_type_id] |= sponsor
+
+        for sponsor_category, sponsors in sponsor_categories_dict.items():
```

Comment:
Not sure this intermediate dict is required, seems we could do without it with list/dict comprehension

Reply:
Related to above point, it's used for showing random sponsors but the code was lost.